pkg/segment/segment.go

```diff
@@ -67,12 +69,18 @@ func NewClient(preference *preference.PreferenceInfo) (*Client, error) {
 
 // newCustomClient returns a Client created with custom args
 func newCustomClient(preference *preference.PreferenceInfo, telemetryFilePath string, segmentEndpoint string) (*Client, error) {
+	// get the locale information
+	tag, err := locale.Detect()
+	if err != nil {
+		return nil, err
```

Comment:
If locale detection fails, the whole telemetry will fail, and nothing will be send, do we want that?

Reply:
Snap. I did it correctly when I added locale info in `traits`. Fixed now.